src/app/pages/categories/management/categories-editor/pages-categories-editor.component.html

```diff
@@ -0,0 +1,33 @@
+<form [formGroup]="form" class="form__container">
+  <h1 mat-dialog-title>{{ data.id ? "Update Category" : "Create Category" }}</h1>
+  <mat-form-field>
+    <mat-label>Select type</mat-label>
+    <div class="loading-container" *ngIf="transactionTypeDisabledLoading">
```

Comment:
I would make some changes that will simplify this:
- Hide the whole form and show loading indicator centered in modal
- Do not use tooltip but gray text under disabled input so user can know as soon as he opens the modal that editing type is blocked -> instead of hovering over select to show tooltip